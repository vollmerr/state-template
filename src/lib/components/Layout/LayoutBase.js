import React from 'react';
import classNames from 'classnames';
import T from 'prop-types';

const LayoutBase = (props) => {
  const {
    baseClass,
    children,
    className,
    tag: Tag,
    ...rest
  } = props;

  const cn = classNames([
    baseClass,
    className,
  ]);

  return (
    <Tag className={cn} {...rest}>
      {children}
    </Tag>
  );
};

LayoutBase.propTypes = {
  baseClass: T.string,
  children: T.node.isRequired,
  className: T.string,
  tag: T.node,
};

LayoutBase.defaultProps = {
  baseClass: '',
  className: '',
  tag: 'div',
};

export default LayoutBase;
