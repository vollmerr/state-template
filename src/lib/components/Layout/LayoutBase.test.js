import React from 'react';
import { shallow } from 'enzyme';

import { LayoutBase as fromLib } from '../..';
import LayoutBase from './LayoutBase';

const props = {
  baseClass: 'test-baseclass',
  children: 'test-children',
  className: 'test-class',
  tag: 'div',
};

let wrapper;
describe('LayoutBase', () => {
  beforeEach(() => {
    wrapper = shallow(<LayoutBase {...props} />);
  });

  it('should be exported as part of the library', () => {
    expect(fromLib).toEqual(LayoutBase);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should attach class names to the base class', () => {
    expect(wrapper.prop('className')).toMatch(props.baseClass);
    expect(wrapper.prop('className')).toMatch(props.className);
  });

  it('should pass additional props to the tag', () => {
    wrapper.setProps({ id: 'test-id' });
    expect(wrapper.find({ id: 'test-id' }).length).toBe(1);
  });

  it('should render as the html tag provided', () => {
    wrapper.setProps({ tag: 'span' });
    expect(wrapper.find('span').length).toBe(1);
  });
});
