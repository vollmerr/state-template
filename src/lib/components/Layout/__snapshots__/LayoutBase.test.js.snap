// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutBase should render correctly 1`] = `
<div
  className="test-baseclass test-class"
>
  test-children
</div>
`;
